gcloud-java-logging/src/test/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2ApiTest.java

```diff
@@ -0,0 +1,132 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.gcloud.logging.spi.v2;
+
+import com.google.api.MonitoredResource;
+import com.google.gcloud.logging.testing.LocalLoggingHelper;
+import com.google.logging.v2.LogEntry;
+
+import io.gapi.gax.grpc.ServiceApiSettings;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class LoggingServiceV2ApiTest {
+  private static LocalLoggingHelper loggingHelper;
+  private LoggingServiceV2Api loggingApi;
+
+  @BeforeClass
+  public static void startStaticServer() {
+    loggingHelper = new LocalLoggingHelper("in-process-1");
+    loggingHelper.start();
+  }
+
+  @AfterClass
+  public static void stopServer() {
+    loggingHelper.shutdownNow();
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    loggingHelper.reset();
+    ServiceApiSettings settings = new ServiceApiSettings();
+    settings.setChannel(loggingHelper.createChannel());
+    loggingApi = LoggingServiceV2Api.create(settings);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    loggingApi.close();
+  }
+
+  @Test
+  public void testWriteLog() throws Exception {
+    String logName = LoggingServiceV2Api.createLogNamePath("my-project", "my-log");
+    MonitoredResource resource = MonitoredResource.newBuilder().build();
+    List<LogEntry> entries = new ArrayList<>();
+    entries.add(LogEntry.newBuilder()
+        .setLogName(logName)
+        .setTextPayload("foobar")
+        .build()
+    );
+    loggingApi.writeLogEntries(logName, resource, Collections.<String, String>emptyMap(), entries);
+  }
+
+  @Test
+  public void testListLog() throws Exception {
+    String logName = LoggingServiceV2Api.createLogNamePath("my-project", "my-log");
+    MonitoredResource resource = MonitoredResource.newBuilder().build();
+    List<LogEntry> entries = new ArrayList<>();
+    entries.add(LogEntry.newBuilder()
+        .setLogName(logName)
+        .setTextPayload("foobar")
+        .build()
+    );
+    loggingApi.writeLogEntries(logName, resource, Collections.<String, String>emptyMap(), entries);
+
+    Iterable<LogEntry> gotEntries = loggingApi.listLogEntries(
+        Collections.singletonList("my-project"), "", ""
+    );
+    List<LogEntry> gotList = new ArrayList<>();
+    for (LogEntry e : gotEntries) {
+      gotList.add(e);
+    }
+    Assert.assertEquals(entries, gotList);
+  }
+
+  @Test
```

Comment:
As noted above, this should not be part of #472 (unless you plan to remove this code and wait until that issue is resolved).